cmd/swarm/upload.go

```diff
@@ -159,11 +180,17 @@ func (c *client) uploadDirectory(dir string, defaultPath string) (string, error)
 	return mhash, err
 }
 
-func (c *client) uploadFile(file string, fi os.FileInfo) (manifestEntry, error) {
+func (c *client) uploadFile(file string, fi os.FileInfo, mimetype_ string) (manifestEntry, error) {
+	var mimetype string
 	hash, err := c.uploadFileContent(file, fi)
+	if mimetype_ != "" {
+		mimetype = mimetype_
+	} else {
```

Comment:
of course it is still nice to have mime-type as an option to custom override it.
great to see this option also available for file uploads. 
This should be documented though